src/lender/reserve.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity >=0.5.15 <0.6.0;
+
+import "tinlake-math/math.sol";
+import "tinlake-auth/auth.sol";
+
+contract ERC20Like {
+  function balanceOf(address) public view returns (uint);
+  function transferFrom(address,address,uint) public returns (bool);
+  function mint(address, uint) public;
+  function burn(address, uint) public;
+  function totalSupply() public view returns (uint);
+}
+
+contract ShelfLike {
+    function balanceRequest() public returns (bool requestWant, uint amount);
+}
+
+contract Reserve is Math, Auth {
+    ERC20Like public currency;
+    ShelfLike public shelf;
+
+    bool public poolActive;
+    uint256 public currencyAvailable;
+
+    address self;
+
+    constructor(address currency_) public {
+        wards[msg.sender] = 1;
+        currency = ERC20Like(currency_);
+        poolActive = true;
+        self = address(this);
+    }
+
+    function depend(bytes32 contractName, address addr) public auth {
+        if (contractName == "shelf") { shelf = ShelfLike(addr); }
+        else if (contractName == "currency") { currency = ERC20Like(addr); }
+        else revert();
+    }
+
+    function activatePool(bool active) public auth {
```

Comment:
I think we don't need the activatePool method. i could just use `updateMaxCurrency(0)`. For the same effect

Reply:
Because we only want to deactivate borrows. Repay are still okay